Update Haskell package set (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2020-11-06 20:00 +02:00. You can watch this live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR that viewers can use to chat with me and with each other.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-27-g11a103b from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/6ce5b767941b7361d0db86fdb076b7ffe64e07b2.

KaiHa committed:
haskellPackages.git-mediate: mark as not broken

After update to 1.0.8.1 git-mediate is buildable again

cdepillabout committed:
haskellPackages.shellFor: improve documentation

cdepillabout committed:
haskellPackages.shellFor: fix spelling and grammar mistakes

Co-authored-by: maralorn <malte.brandy@maralorn.de>

cdepillabout committed:
haskellPackages.shellFor: add a doBenchmark argument for enabling benchmarks deps

expipiplus1 committed:
haskellPackages.vulkan-utils: Add necessary dependency on vulkan-headers

Also mark as unbroken

expipiplus1 committed:
haskellPackages.autoapply: Depend on correct version of th-desugar

Also mark as unbroken

expipiplus1 committed:
haskellPackages.exact-real: mark unbroken

expipiplus1 committed:
haskellPackages.orbits: mark unbroken

peti committed:
haskell-gi-base: update override for the new version

peti commented:
No LTS update this week. https://www.stackage.org/ is still at LTS 16.20.

peti committed:
pandoc: update override for the new version

peti committed:
haskell-configuration-common: cosmetic

peti committed:
all-cabal-hashes: update to Hackage at 2020-11-06T01:07:40Z

peti committed:
hackage2nix: update list of broken builds to fix evaluation errors on Hydra

Ping @cdepillabout because of broken spago build.

maralorn committed:
haskellPackages.dependent-sum-aeson-orphans: Remove obsolete patch

expipiplus1 committed:
haskellPackages.vulkan: add 32 bit platforms to unsupported-platforms

Same for haskellPackages.VulkanMemoryAllocator

expipiplus1 commented:
Is it possible to get https://github.com/NixOS/nixpkgs/pull/102997 included this week? I'm going to sleep now, but happy for any necessary changes to be made :D

peti committed:
git-annex: update sha256 hash for new version 8.20201103

peti committed:
ghcjs: mark build broken

peti committed:
haskell-binary-instances: drop obsolete override

Unfortunately, this doesn't mean that the package now compiles
because it still depends on an newer version of time-compat. :-(

peti committed:
haskell: drop obsolete patches for patch and reflex

peti commented:
Next week we'll merge https://github.com/NixOS/nixpkgs/pull/103038.
